src/StudyEntities/Candidate/SiteIDGenerator.php

```diff
@@ -0,0 +1,256 @@
+<?php declare(strict_types=1);
+
+namespace LORIS\StudyEntities\Candidate;
+
+class SiteIDGenerator extends IdentifierGenerator
+{
+    /* Either 'PSCID' or 'ExternalID' */
+    private const LENGTH     = 4;
+    protected $kind;
+
+    /**
+     * Creates a new instance of a SiteIDGenerator to create either PSCIDs or
+     * ExternalIDs. Relevant properties are extracted from the config.xml file.
+     */
+    public function __construct(?string $prefix = null)
+    {
+        // Read config settings from project/config.xml to retrieve the
+        // alphabet, length, and generation method (sequential or random) used
+        // to create new IDs.
+        $this->generationMethod = $this->getIDSetting('generation');
+        $this->length = $this->getIDSetting('length') ?? self::LENGTH;
+        $this->alphabet = $this->getIDSetting('alphabet');
+        // Initialize minimum and maximum allowed values for IDs. Set the values
+        // to the lowest/highest character in $alphabet repeated $length times
+        // if the min or max is not configured in project/config.xml
+        $this->minValue         = $this->getIDSetting('min') ??
+            str_repeat(strval($this->alphabet[0]), $this->length);
+        $this->maxValue         = $this->getIDSetting('max') ??
+            str_repeat(
+                strval($this->alphabet[count($this->alphabet) - 1]), 
+                $this->length
+            );
+        $this->prefix = $prefix ?? $this->getIDSetting('prefix');
+        $this->validate();
+    }
+
+    protected function validate(): void {
+        if (empty($this->generationMethod)
+            || empty($this->length)
+            || empty($this->alphabet)
+            || empty($this->minValue)
+            || empty($this->maxValue)
+            || empty($this->prefix)
+            || !is_array($this->alphabet)
+        ) {
+            throw new \DomainException(
+                'Values not configured properly for ' . get_class($this) . '. '
+                . 'Please correct your configuration file.'
+                . "Length: `{$this->length}`\n"
+                . "Alphabet: `" . implode($this->alphabet) . "`\n"
+                . "Min: `{$this->minValue}`\n"
+                . "Max: `{$this->maxValue}`\n"
+                . "Length: `{$this->prefix}`\n"
+                . "Generation: `{$this->generationMethod}`\n"
+            );
+        }
+        if ($this->kind !== 'PSCID' && $this->kind !== 'ExternalID') {
```

Comment:
I think we could skip this check. If a config.xml contains infos about an other kind of ID, either validate of getExistingIds will fail if it is not configured.